src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -131,6 +131,15 @@ public static PhysicalAddress Parse(string address)
             {
                 hasDashes = true;
                 buffer = new byte[(address.Length + 1) / 3];
+
+                string[] parts = address.Split('-');
+                for (int i = 0; i < parts.Length; i++)
+                {
+                    if (parts[i].Length < 2)
+                    {
+                        throw new FormatException(SR.net_bad_mac_address);
+                    }
+                }
```

Comment:
I think @karelz's suggestion was essentially just to do:
```C#
if ((address.Length + 1) % 3 != 0)
{
    throw new FormatException(SR.net_bad_mac_address);
}
```
here.  Does that address the problem?

It would also be good to add tests for Parse.  Right now the tests for PhysicalAddress are fairly sparse https://github.com/dotnet/corefx/blob/master/src/System.Net.NetworkInformation/tests/FunctionalTests/PhysicalAddressTest.cs and don't include any for Parse.  As part of touching this code, we'll want to make sure it continues to properly parse ones expected to be valid and fail on ones expected to be invalid, e.g.
```C#
[Theory]
[InlineData("somevalidaddress", new byte[] { expectedaddressbytes }
... // more inline datas
public void Parse_Valid_Success(string input, byte[] expected)
{
    ... // validate that Parse(input) produces expected
}

[Theory]
[InlineData("someinvalidaddress")]
... // more inline datas
public void Parse_Invalid_ThrowsFormatException(string input)
{
    ... // validate that Parse(input) throws a FormatException
}
```

Reply:
@Smartis2812 note one reason for the feedback is that what you have allocates unnecessarily.